### ISSUE TITLE
Training of segmentation models

### ISSUE THREAD

**Issue by Synaproot7321** [ultralytics/ultralytics#24343]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi! I’m training a YOLOv8m-seg model and I've included 10% background images in my dataset, as suggested in the docs. However, I’m still getting some warnings. Does the 10% 'no-detection' recommendation apply to both object detection and segmentation, or is it just for detection?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Synaproot7321, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue moving quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a custom training ❓ question related to segmentation training, please provide as much information as possible, including dataset image examples, label format details, training logs, the exact warnings you are seeing, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). If you believe the warnings indicate a bug 🐛, please also provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Synaproot7321** [comment]
@UltralyticsAbi @UltralyticsAssistant @matt-ultralytics

**Comment by Y-T-G** [comment]
Hello.

Can you elaborate what warnings you're receiving?

**Comment by Synaproot7321** [comment]
Thank for your answer, this is the warning I'm getting:

optimizer: AdamW(lr=0.001429, momentum=0.9) with parameter groups 86 weight(decay=0.0), 97 weight(decay=0.0005), 96 bias(decay=0.0)
Plotting labels to C:\Users\leona\Documents\Projects\Sistema de monitoreo del estado de desgaste de fajas transportadoras\demo\TRAIN\runs\segment\faja_seg_v1\labels.jpg... 
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to C:\Users\leona\Documents\Projects\Sistema de monitoreo del estado de desgaste de fajas transportadoras\demo\TRAIN\runs\segment\faja_seg_v1
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss   sem_loss  Instances       Size
       1/50      7.77G      2.198      2.312      7.235     0.8703          0          1        640: 100% ━━━━━━━━━━━━ 44/44 13.0s/it 9:34
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 1.2it/s 0.8s
C:\Users\leona\Documents\Projects\Sistema de monitoreo del estado de desgaste de fajas transportadoras\demo\env\Lib\site-packages\ultralytics\utils\metrics.py:839: RuntimeWarning: Mean of empty slice
  i = smooth(f1_curve.mean(0), 0.1).argmax()  # max F1 index
C:\Users\leona\Documents\Projects\Sistema de monitoreo del estado de desgaste de fajas transportadoras\demo\env\Lib\site-packages\numpy\_core\_methods.py:134: RuntimeWarning: invalid value encountered in divide
  ret = um.true_divide(
                   all          3          0          0          0          0          0          0          0          0          0
WARNING no labels found in segment set, cannot compute metrics without labels

**Comment by Y-T-G** [comment]
There should be more warnings before that log.

Can you post the logs from the beginning?

**Comment by Synaproot7321** [comment]
Hi, I don't have the rest of the log, but could you tell me if you usually include images without detections in your training or validation datasets when using YOLO segmentation? If so, what percentage do you recommend in both?

**Comment by glenn-jocher** [comment]
Yes, background-only images are fine for segmentation too, but there’s no strict target percentage; a small amount in `train` is okay if it matches your real data, while `val` should not be mostly or entirely empty. Your warning is not from the 10% itself—it means YOLO found `0` segment instances in `val`, so please check that your labels follow the [segmentation format](https://docs.ultralytics.com/datasets/segment/) and then delete any `labels/*.cache` files after fixing them, since mixed or invalid box/segment labels can cause segments to be dropped during loading as shown in [`YOLODataset.get_labels()`](https://docs.ultralytics.com/reference/data/dataset/).
